src/modules/cpu/kernel/fog.hpp

```diff
@@ -233,21 +259,108 @@ RppStatus fog_u8_u8_host_tensor(Rpp8u *srcPtr,
 #endif
                 for (; vectorLoopCount < bufferLength; vectorLoopCount += 3)
                 {
+                    RpptFloatRGB pixel;
+                    pixel.R = static_cast<Rpp32f>(srcPtrTemp[0]);
+                    pixel.G = static_cast<Rpp32f>(srcPtrTemp[1]);
+                    pixel.B = static_cast<Rpp32f>(srcPtrTemp[2]);
+                    Rpp32f gray = 0.299 * pixel.R + 0.587 * pixel.G + 0.114 * pixel.B;
+                    gray = gray * grayValue;
+                    Rpp32f oneMinusGrayValue = 1 - grayValue;
+                    pixel.R = (pixel.R * oneMinusGrayValue + gray);   
+                    pixel.G = (pixel.G * oneMinusGrayValue + gray);   
+                    pixel.B = (pixel.B * oneMinusGrayValue + gray);  
```

Comment:
Perhaps consolidate all these lines to the below. I've also changed the close-parentheses position on the last 3 lines.
```
                    RpptFloatRGB pixel = {static_cast<Rpp32f>(srcPtrTemp[0]),
                                          static_cast<Rpp32f>(srcPtrTemp[1])
                                          static_cast<Rpp32f>(srcPtrTemp[2])};
                    Rpp32f gray = grayValue * ((RGB_TO_GREY_WEIGHT_RED * pixel.R) + (RGB_TO_GREY_WEIGHT_GREEN * pixel.G) + (RGB_TO_GREY_WEIGHT_BLUE * pixel.B));
                    Rpp32f oneMinusGrayValue = 1 - grayValue;
                    pixel.R = (pixel.R * oneMinusGrayValue) + gray;
                    pixel.G = (pixel.G * oneMinusGrayValue) + gray;
                    pixel.B = (pixel.B * oneMinusGrayValue) + gray;
```